src/main/java/de/hysky/skyblocker/skyblock/itemlist/ItemStackBuilder.java

```diff
@@ -31,73 +41,60 @@ public static ItemStack fromNEUItem(NEUItem item) {
 
         List<Pair<String, String>> injectors = new ArrayList<>(petData(internalName));
 
-        NbtCompound root = new NbtCompound();
-        root.put("Count", NbtByte.of((byte) 1));
+        String legacyId = item.getMinecraftItemId();
+        Identifier itemId = new Identifier(ItemFixerUpper.convertItemId(legacyId, item.getDamage()));
+        
+        ItemStack stack = new ItemStack(Registries.ITEM.get(itemId));
 
-        String id = item.getMinecraftItemId();
-        int damage = item.getDamage();
-        root.put("id", NbtString.of(ItemFixerUpper.convertItemId(id, damage)));
+        // Custom Data
+        NbtCompound customData = new NbtCompound();
 
-        NbtCompound tag = new NbtCompound();
-        root.put("tag", tag);
-
-        NbtCompound extra = new NbtCompound();
-        tag.put(ItemUtils.EXTRA_ATTRIBUTES, extra);
-        extra.put(ItemUtils.ID, NbtString.of(internalName));
-
-        NbtCompound display = new NbtCompound();
-        tag.put("display", display);
+        // Add Skyblock Item Id
+        customData.put(ItemUtils.ID, NbtString.of(internalName));
 
+        // Item Name
         String name = injectData(item.getDisplayName(), injectors);
-        display.put("Name", NbtString.of(Text.Serialization.toJsonString(Text.of(name))));
+        stack.set(DataComponentTypes.CUSTOM_NAME, Text.of(name));
 
-        NbtList lore = new NbtList();
-        display.put("Lore", lore);
-        item.getLore().forEach(el -> lore.add(NbtString.of(Text.Serialization.toJsonString(Text.of(injectData(el, injectors))))));
+        // Lore
+        stack.set(DataComponentTypes.LORE, new LoreComponent(item.getLore().stream().map(line -> Text.of(injectData(line, injectors))).collect(Collectors.toCollection(() -> new ArrayList<>()))));
```

Comment:
Is there a reason for this to be a mutable list?

Reply:
Don’t think so, I was just trying to fix things.